### PR TITLE
Dedup fix

### PR DESCRIPTION
**Pull request by JudahSchwartz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by YuvHayun — APPROVED** [review]

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 1]
```diff
@@ -1098,40 +1111,41 @@ def fetch_incidents(client, first_fetch_time, integration_instance, exclude_arti
...
-                gte_creation_time_milliseconds=last_fetch, limit=max_fetch,
-                starred=starred,
-                starred_incidents_fetch_window=starred_incidents_fetch_window,
-                exclude_artifacts=exclude_artifacts)
+            deduped_incidents.append(raw_incident)
 
-    # demisto.debug(f"{raw_incidents=}") # uncomment to debug, otherwise spams the log
+    if len_events:= len(deduped_incidents) > max_fetch:
+        demisto.debug(f'len(deduped_incidents)={len_events}')
+        deduped_incidents = deduped_incidents[:max_fetch]
```
[the remark this answers is not in the document]
Comment (JudahSchwartz) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1138]: want to simplify, we'll bring them next time

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 2]
```diff
@@ -1142,44 +1156,53 @@ def fetch_incidents(client, first_fetch_time, integration_instance, exclude_arti
...
             if demisto.params().get('sync_owners') and incident_data.get('assigned_user_mail'):
-                incident['owner'] = demisto.findUser(email=incident_data.get('assigned_user_mail')).get('username')
+                incident['owner'] = demisto.findUser(email=incident_data['assigned_user_mail']).get('username')
             # Update last run and add incident if the incident is newer than last fetch
-            if incident_data.get('creation_time', 0) > last_fetch:
+            creation_time = incident_data.get('creation_time', 0)
+            demisto.debug(f'creation time for {incident_id=} {creation_time=}')
+            if creation_time > last_fetch:
+                next_dedup_incidents = [incident_id]
+                demisto.debug(f'updating last_fetch,  {incident_id=}')
```
[the remark this answers is not in the document]
Comment (JudahSchwartz) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1178]: its printed above

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py [hunk 3]
```diff
@@ -428,20 +428,19 @@ def test_get_remote_data_command_with_rate_limit_exception(mocker):
...
+    import sys
     client = Client(
         base_url=f'{XDR_URL}/public_api/v1', verify=False, timeout=120, proxy=False)
     args = {
         'id': 1,
         'lastUpdate': 0
     }
 
-    mocker.patch.object(demisto, 'results')
+    mocker.patch('CortexXDRIR.return_error', side_effect=sys.exit)
```
Comment (JudahSchwartz) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py:439]: what?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 4]
```diff
@@ -1098,40 +1111,33 @@ def fetch_incidents(client, first_fetch_time, integration_instance, exclude_arti
         demisto.debug(
             f"starred_incidents_fetch_window after parsing date range {starred_incidents_fetch_window}")
 
-    incidents = []
     if incidents_from_previous_run:
+        demisto.debug('Using incidents from last run')
```
Comment (YuvHayun) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1115]: I thought we want to remove this

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py (hunk 3 above)
Comment (jlevypaloalto) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR_test.py:439]: Reverted to the version on master

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py (hunk 4 above)
Comment (jlevypaloalto) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1115]: We removed this from common use, but it's still used when there is a rate-limit whilst enriching incidents with >50 alerts

**Pull request merged**